### ISSUE TITLE
New `fluent` sink

### ISSUE THREAD

**Issue by binarylogic** [vectordotdev/vector#534]
This will make it easier for users to transition in production environments. We basically want to support the `forward` input and output for Fluentbit. I believe it's the same protocol used for FluentD, but we should investigate further.  They have [a section](https://docs.fluentbit.io/manual/output/forward#fluentd) in their docs that covers this.

**Comment by szibis** [comment]
Is there any timeline for this and fluent source https://github.com/timberio/vector/issues/419. This may help to adopt Vector in systems where Fluentd is the main platform and replace part by part with Vector.

**Comment by sichvoge** [comment]
Any updates?

**Comment by jszwedko** [comment]
Hey @sichvoge ! Nothing yet on the sink side, but we did recently add support for receiving messages via fluent's forward protocol in #7548 .

**Comment by szibis** [comment]
@jszwedko yes source is great for fluentd but many people already have fluentd integrated into many places especially when using Kubernetes and also many use Fluentd forwarder to move data for example into external systems that understand that.
In many cases, there is no option to add Vector before those solutions that already integrated with Fleuntd that's why simulating fluentd as a sender is the best option to integrate with. 
Vector needs to face already some legacy in modern systems especially that fluentd in Kubernetes is/was the primary solution used and to push Vector as replacement we need to have all possible option here to cover.

**Comment by alanconway** [comment]
Is this issue still alive? We're moving our logging operator from fluentd to vector, we need to be able to forward _to_ fluentd to have feature parity and avoid breaking existing customers that are doing this. Fluentd is a pretty common legacy choice as "middle man" to broker log formats, sometimes crossing organizational boundaries. It would help adoption of vector if it was possible to upgrade such systems one step at a time, even if the first step is _sending_ to a remote fluetnd.

**Comment by spencergilbert** [comment]
Still alive but not actively prioritized at this time - we'd be happy to accept a PR of course!

**Comment by jszwedko** [comment]
Hey @alanconway ! We don't have any immediate plans for this (though would happily accept a PR!) but one work around is to use Vector's `http` sink with FluentD's `http` source rather than the native fluent forward protocol.

**Comment by alanconway** [comment]
Thanks @jszwedko , that is the same workaround I had in mind.

**Comment by ganga1980** [comment]
Hi, @jszwedko , are there any plans/wip for adding fluent sink?   @alanconway , Did you try out vector http sink to fluent http source, did you see any perf & throughput differences (fluent output forward to fluent input forward) ?

**Comment by ffcactus** [comment]
Hopefully, we will have fluentd sink soon.

**Comment by hex-m** [comment]
Version 1.5 of the [protocol specification](https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1.5) was released in the meantime.

**Comment by voronin-ilya** [comment]
In the meantime, we were able to pair Vector and Fluent Bit using an HTTP sink.

vector sink configuration:
```yaml
sinks:
  http:
    type: http
    inputs: ["your-input"]
    uri: http://fluent-bit-host:9880
    batch:
      max_events: 1
    encoding:
      codec: native_json
    request:
      headers:
        Content-Type: application/json
```

Relevant fluent-bit config snippet:
```ini
[INPUT]
    name http
```

Most likely, that design won’t be able to handle high event count workloads, but it works well for low-volume event pipelines.
